File: src/main/kotlin/org/kite9/diagram/common/elements/edge/BiDirectionalPlanarizationEdge.java
package org.kite9.diagram.common.elements.edge;

import org.kite9.diagram.model.Connected;
import org.kite9.diagram.model.DiagramElement;

public interface BiDirectionalPlanarizationEdge extends PlanarizationEdge {

	public DiagramElement getOriginalUnderlying();
	
	public Connected getFromConnected();
	
	public Connected getToConnected();
}

File: src/main/kotlin/org/kite9/diagram/common/elements/edge/BiDirectionalPlanarizationEdge.kt
package org.kite9.diagram.common.elements.edge

import org.kite9.diagram.model.Connected
import org.kite9.diagram.model.DiagramElement

interface BiDirectionalPlanarizationEdge : PlanarizationEdge {

    fun getOriginalUnderlying(): DiagramElement
    fun getFromConnected(): Connected
    fun getToConnected(): Connected
}